pkg/apis/resource/validation/validation.go

```diff
@@ -433,179 +512,244 @@ func validateNodeName(name string, fldPath *field.Path) field.ErrorList {
 }
 
 // ValidateResourceSlice tests if a ResourceSlice object is valid.
-func ValidateResourceSlice(resourceSlice *resource.ResourceSlice) field.ErrorList {
-	allErrs := corevalidation.ValidateObjectMeta(&resourceSlice.ObjectMeta, false, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
-	if resourceSlice.NodeName != "" {
-		allErrs = append(allErrs, validateNodeName(resourceSlice.NodeName, field.NewPath("nodeName"))...)
-	}
-	allErrs = append(allErrs, validateResourceDriverName(resourceSlice.DriverName, field.NewPath("driverName"))...)
-	allErrs = append(allErrs, validateResourceModel(&resourceSlice.ResourceModel, nil)...)
+func ValidateResourceSlice(slice *resource.ResourceSlice) field.ErrorList {
+	allErrs := corevalidation.ValidateObjectMeta(&slice.ObjectMeta, false, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
```

Comment:
This is not user-facing and is in practice (= `resourceslicecontroller.go`) a generated name:
```
GenerateName: c.owner.Name + "-" + c.driver + "-",
```

That code will have to become a bit more careful about replacing slash with hyphen, otherwise the base name might not be a DNS label.

But that's doable, so yes. I'll simplify it.

Reply:
Isn't the drive name a DNS subdomain though? How can we construct a DNS label out of
```
GenerateName: c.owner.Name + "-" + c.driver + "-",
```
if `c.driver` is a DNS subdomain?

Reply:
That's what I meant with "replacing slash with hyphen" - oops, I meant "dot". Here's what the new code would have to look like:
```
GenerateName: strings.ReplaceAll(c.driver + "-" + c.owner.Name + "-", ".", "-"),
```

Note that this puts the driver name first because it's likely to be shorter. I remember some clusters have really long generated node names and that part is less important.

Reply:
As I am making this change, I keep finding places where some other name is used as ResourceSlice name. For example, the unique namespace name is used also for a unique per-test ResourceSlice in one E2E test which simulates network-attached devices.

I'm not seeing any other usage of NameIsDNSLabel besides ValidateNamespaceName. Should we perhaps stick with the NameIsDNSSubdomain for consistency with pretty much all other API resources? Or am I missing some other cases?

Reply:
I am fine with it, I just want to make sure we are not just copying patterns for no good reason.  Non-namespaces objects tend to want more unique names, and that is fine.